From cd8c1c64acedacb98214763352ffe5dbe1a0ff0b Mon Sep 17 00:00:00 2001
From: Afzal Mohammed <afzal@ti.com>
Date: Thu, 21 Nov 2013 16:01:33 +0530
Subject: [PATCH 0663/1115] ARM: proc-v7: avoid ldm access crossing 1K
 boundary

Prevent ldm access crossing 1K page boundary by aligning relevant
section to 1K. AM43x hangs if ldm access crosses 1K page boundary. This
issue was hit with a particular combination of CONFIG options, presently
with CONFIG_PREEMPT enabled and CONFIG_DEBUG_LL disabled.

Hang is due to bug in early internal version of AM43x engineering
sample, production version does not have this bug and can be reverted
at that time.

Problem here is similar to as that mentioned in
"3ca5009 ARM: head.S: avoid ldm access crossing 1K page boundary".

Signed-off-by: Afzal Mohammed <afzal@ti.com>
(cherry picked from commit 86617f774ab0426771046e329854fb605b285ffa)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/mm/proc-v7.S |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mm/proc-v7.S b/arch/arm/mm/proc-v7.S
index 338a386..f9699e4 100644
--- a/arch/arm/mm/proc-v7.S
+++ b/arch/arm/mm/proc-v7.S
@@ -379,7 +379,7 @@ __v7_setup:
 	mov	pc, lr				@ return to head.S:__ret
 ENDPROC(__v7_setup)
 
-	.align	2
+	.align	10
 __v7_setup_stack:
 	.space	4 * 11				@ 11 registers
 
-- 
1.7.5.4

